onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -124,10 +129,142 @@ def _parse_line(line):
+        parameter width: image width in pixels
+        return: matrix characterizing image
+    '''
+    pillow_img = Image.open(image_filepath).resize((width, height))
+    input_data = np.float32(pillow_img)/127.5 - 1.0 # normalization
+    input_data -= np.mean(input_data)
+    nhwc_data = np.expand_dims(input_data, axis=0)
+    nchw_data = nhwc_data.transpose(0, 3, 1, 2) # ONNX Runtime standard
+    return nchw_data
+
+def load_batch(images_folder, height, width, size_limit=30):
+    '''
+    Loads a batch of images
+    parameter images_folder: path to folder storing images
+    parameter height: image height in pixels
+    parameter width: image width in pixels
+    parameter size_limit: number of images used to run inference
+    return: list of matrices characterizing multiple images
+    '''
+    image_names = os.listdir(images_folder)
+    if len(image_names) >= size_limit:
+        batch_filenames = [image_names[i] for i in range(size_limit)]
+    else:
+        batch_filenames = image_names
+    unconcatenated_batch_data = []
+    for image_name in batch_filenames:
+        image_filepath = images_folder + '/' + image_name
+        nchw_data = load_and_resize_image(image_filepath, height, width)
+        unconcatenated_batch_data.append(nchw_data)
+    batch_data = np.concatenate(np.expand_dims(unconcatenated_batch_data, axis=0), axis=0)
+    return batch_data
+
+# Using augmented outputs to generate inputs to quantize.py
+def get_intermediate_outputs(augmented_model_path, session, inputs, calib_mode='naive'):
+    '''
+    Gather intermediate model outputs after running inference
+        parameter model: path to augmented FP32 ONNX model
+        parameter inputs: list of loaded test inputs (or image matrices)
+        parameter average_mode: type 'naive' gives (ReduceMin, ReduceMax) pairs
+                                for each augmented node across test data sets, where
+                                the first element is a minimum of all ReduceMin values
+                                and the second element is a maximum of all ReduceMax
+                                values; the type 'smooth' yields a smooth average
+                                with anomalous values removed
+        return: dictionary mapping added node names to (ReduceMin, ReduceMax) pairs
+    '''
+    num_inputs = len(inputs)
+    input_name = session.get_inputs()[0].name
+    intermediate_outputs = [session.run([], {input_name: inputs[i]}) for i in range(num_inputs)]
+
+    # Creating dictionary with output results from multiple test inputs
+    output_node_names = [session.get_outputs()[i].name for i in range(len(intermediate_outputs[0]))]
+    output_dicts = [dict(zip(output_node_names, intermediate_outputs[i])) for i in range(num_inputs)]
+    merged_dict = {}
+    for d in output_dicts:
+        for k, v in d.items():
+            merged_dict.setdefault(k, []).append(v)
+    added_output_node_names = output_node_names[1:]
```

Comment:
output_node_names[1:]
here you are assuming original model has only 1 output... we cannot make this assumption... you should find the number of outputs from the original model

Reply:
Fixed.